### PR TITLE
Updates regform.html to remove line of code that makes the staffing checkbox readonly when an attendee has shifts

### PR DESCRIPTION
**Pull request by RobRuana**

### CODE REVIEW

File: uber/templates/regform.html [hunk 1]
```diff
@@ -92,7 +92,6 @@
     $(function () {
         staffingClicked();
         if ($.field('staffing')) {
-            {% if attendee.shifts %} $.field('staffing').prop('readonly', true).prop('title', 'Please {{ "contact your department head to drop shifts" if c.AT_THE_CON else "unassign yourself from shifts" }} before changing your volunteer status.'); {% endif %}
```
Comment (kitsuta) [inline uber/templates/regform.html]: This is fine, but do you think it's better to just put this line behind a `{% if not admin_area %}` check? That way the checkbox behaves as we want it to on the public pages.

Comment (RobRuana) [inline uber/templates/regform.html]: Sure. If we do that, should we also add back the other deleted line?
```python
{% if c.AT_THE_CON %} $.field('staffing').prop('disabled', true).prop('title', 'Please see Staffing Operations to change your volunteer status.'); {% endif %}
```

Comment (RobRuana) [inline uber/templates/regform.html]: This is interesting, a detail I knew in the deep recesses of my mind, but completely forgot about. Readonly checkboxes aren't actually readonly, you can still click and change them because you aren't changing their `value`, only their `checked` state.

You can make a checkbox actually readonly by setting its `disabled` property, but then it won't be posted to the server (this is the bug that was deleting everyone's shifts).

Instead of doing this in javascript, I think the correct solution is to render a disabled checkbox & a hidden input that will actually post the value.

**Comment by kitsuta** [comment]
Ohh, I had no idea! That's super tricky. We can just remove it for now,
then.

File: uber/templates/macros.html [hunk 2]
```diff
@@ -21,9 +21,33 @@
...
+        {% if model[field_name] %}checked{% endif %} />
+    {% else %}
+      <input
+        type="checkbox"
+        name="{{ field_name }}"
+        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+        value="1"
+        {% if not label and title %}title="{{ title }}"{% endif %}
+        {% if model[field_name] %}checked{% endif %} />
+    {% endif %}
```
Comment (RobRuana) [inline uber/templates/macros.html]: I updated our checkbox macro to support readonly checkboxes. If `readonly=True` is passed, then a disabled checkbox will be rendered with a hidden input for the actual field value. In this manner we can create readonly checkboxes, while retaining the ability to POST the correct field value back to the server.

File: uber/templates/preregistration/attendee_donation_form.html [hunk 3]
```diff
@@ -4,14 +4,15 @@
 {% block content %}
 <div class="masthead"></div>
 <div class="panel panel-default">
-{% if attendee.paid == c.NOT_PAID %}
-    <h2> Badge Payment for {{ attendee.full_name }} </h2>
+  <div class="panel-body">
```
Comment (RobRuana) [inline uber/templates/preregistration/attendee_donation_form.html]: Here I am just fixing a layout issue that I missed in an earlier pull request. Happy to move this to its own pull request, if anyone would like to keep this pull request super specific.

File: uber/templates/regform.html [hunk 4]
```diff
@@ -14,7 +14,8 @@
     $(ageGroupSelected);
 
     var staffingClicked = function () {
-        var checked = $.field('staffing') && $.field('staffing').is(':visible:checked');
+        var $staffingCheckbox = $('label[for="staffing"]').find('input[type="checkbox"]');
+        var checked = $staffingCheckbox && $staffingCheckbox.is(':visible:checked');
```
Comment (RobRuana) [inline uber/templates/regform.html]: Since we now render the staffing checkbox as readonly with a hidden input as the actual field, calling `$.field('staffing')` will not always return the checkbox. If readonly `$.field('staffing')` returns the hidden input. A more general selector is required, `$('label[for="staffing"]').find('input[type="checkbox"]')`.

Although, as I type this, I realize I probably could have used `$('label[for="staffing"] input[type="checkbox"]')`. I can make that change if anyone has a preference.

File: uber/templates/regform.html [hunk 5]
```diff
@@ -357,8 +357,9 @@
 <div class="form-group staffing">
     <label for="staffing" class="col-sm-3 optional-field control-label">Want to Volunteer?</label>
     <div class="checkbox col-sm-6">
-        <label for="staffing">
-        {{ macros.checkbox(attendee, 'staffing') }}
+        {% set staffing_readonly = not admin_area and attendee.shifts %}
+        <label for="staffing" {% if staffing_readonly %}title="Please {{ 'contact your department head to drop shifts' if c.AT_THE_CON else 'unassign yourself from shifts' }} before changing your volunteer status."{% endif %}>
```
Comment (RobRuana) [inline uber/templates/regform.html]: I put the `title` attribute on the `label` instead of on the `checkbox`, so the user gets the hover text no matter where they're hovering.

Comment (kitsuta) [inline uber/templates/regform.html]: Ooh, good catch!

File: uber/templates/regform.html [hunk 6]
```diff
@@ -92,7 +93,6 @@
     $(function () {
         staffingClicked();
         if ($.field('staffing')) {
```
Comment (kitsuta) [inline uber/templates/regform.html]: You changed the selector above to specifically find a checkbox, do we need to do the same here? I don't THINK we do but just wanna check.

Comment (RobRuana) [inline uber/templates/regform.html]: That is a great question! We definitely don't want to look for a checkbox here, because if the checkbox is disabled we don't want it to fire click events (I actually don't know if a disabled checkbox will fire click events, I presume it won't, but it doesn't really matter).

So, `$.field('staffing')` will find the hidden input instead of the checkbox. That should be fine, because you can't click on a hidden input, so it effectively does nothing.

**Comment by RobRuana** [comment]
There were actually two different messages in [this old commit](https://github.com/magfest/ubersystem/blob/60ca6c85dd862921b5221e1103c96f84e8f8bd74/uber/templates/regform.html#L81). I am guessing that was a mistake from the Jinja2 merge.

I must have grabbed the wrong one! I'll make that change.

```python
{% if c.AT_THE_CON %} $.field('staffing').prop('readonly', true).prop('title', 'Please see Staffing Operations to change your volunteer status.'); {% endif %}
{% if attendee.shifts %} $.field('staffing').prop('readonly', true).prop('title', 'Please {{ "contact your department head to drop shifts" if c.AT_THE_CON else "unassign yourself from shifts" }} before changing your volunteer status.'); {% endif %}
```
